consensus/src/aggregation/mod.rs

```diff
@@ -460,10 +492,9 @@ mod tests {
                         engine.start((sender, receiver));
                     }
 
-                    // Create completion watchers for all validators
-                    let mut completion_tasks = Vec::new();
-                    for (validator_pk, mut reporter_mailbox) in reporters.clone() {
-                        let validator = validator_pk.clone();
+                    // Create a single completion watcher for the shared reporter
```

Comment:
Instead of ensuring each validator has all items (not guaranteed by the `aggregator`), we just ensure that a shared reporter has seen all items (and if shutdown happens before it learns of a newly minted certificate, it should hear about on restart via replay).